### ISSUE TITLE
Accessibility Issue:  Labels and radio/checkbox should be connected w/for and id attributes

### ISSUE THREAD

**Issue by rhewitt22** [Leaflet/Leaflet#4797]
I'm running an accessibility audit on my map application and I'm getting errors associated with the layer switcher.  The input is contained within the label element, but I believe you also need to [add a `for` attribute to the label that matches the `id` attribute on the radio/checkbox input](http://a11yproject.com/checklist.html):

**This should pass validation:**
```html
<label for="layer-1">
  <input type="radio" id="layer-1"></input>
</label>
```

I'm using leaflet `v1.0.0-beta.2`.

**Comment by IvanSanchez** [comment]
Nope.

Let me quote from https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label (emphasis mine): 

> The HTML Label Element [...] can be associated with a control ***either*** by placing the control element inside the `<label>` element, ***or*** by using the for attribute.

And your checklist says "the labels must be associated", not "labels must have a `for=` attribute".
